MetaMask/metamask-extension
# Fixing scroll references for Advanced and Security settings

Related: https://github.com/MetaMask/metamask-extension/issues/19479

## Explanation
When anchoring to a settings search result that exists on the Advanced and Security tabs, many of the settings weren't scrolled to correctly, or at all. This is because there was an off by one error in the references ordering in the tab components when compared to the route positions in the settings constants. This PR fixes those off by ones, and in addition for Security constants, standardizes and fixes misspellings in the hash routes.

### Before
(See video in referenced issue)

### After
https://github.com/MetaMask/metamask-extension/assets/8732757/1f145347-66f3-427f-a759-6e582154e9e1

## Manual Testing Steps
1. Navigate to the General Settings tab
2. Using the settings search input, search for and anchor to various advanced and security settings. (`Show balance and token price checker`, `Show hex data`, `Choose your network`, etc)
3. Ensure each setting clicked from in the suggestions is correctly scrolled to. 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

cc: @NiranjanaBinoy
